client/service-worker.js

```diff
@@ -3,7 +3,7 @@
 "use strict";
 
 const cacheName = "__HASH__";
-const excludedPathsFromCache = /^(?:socket\.io|storage|uploads|cdn-cgi)\//;
+const includedPathsInCache = /^(js|css|img|themes|favicon\.ico|fonts|#)\/*/;
```

Comment:
Why are you doing this?
Why can't you just use the same blacklist?

What are you now excluding that was working before that makes your issues go away as far as you are concerned?

Reply:
Some providers (such as for SAML or OpenID) have a redirect URL which can be arbitrary. This would end up being cached by the PWA, causing all kinds of issues. Therefore, using an allowlist instead of a blocklist would make sure that the arbitrary redirect URL is not cached.

Reply:
The login redirect is to the same domain TL is hosted on?
Because if not it should not be an issue as the service worker wouldn't be in scope? Or what am I missing?

Reply:
note to myself, example.com/thelounge is a thing

Reply:
When using SAML or OpenID, there must be a redirect URL configured. This URL must be something that TL isn't using: tl.example.com/r is one example. It can be anything. When authentication occurs, the user is redirected to the provider (ie auth.example.com) which redirects the user back to the app with (tl.example.com/r?callback_data). The PWA can cache this URL which would cause issues with login. Because the callback URL doesn't have to be /r (it can be /redirect, /jfkdkdkd, or anything else), an allowlist would prevent it from being cached.

Reply:
Authentication happens at the reverse proxy level: a cookie is dropped which is checked by the proxy on connection. TL has nothing to do with auth. When using a PWA, it opens an internal web browser to handle the redirect.

Reply:
For HTTP auth to work, the whole of TL must be protected by it. For OpenID, there must be an endpoint that is used for redirect that is unused by the application and also protected by the authentication (so if auth is protecting /tl, then the redirect url must be at /tl/something or /tl/a/b/c)

For reference, here's what the auth looks like when using OpenID:
https://user-images.githubusercontent.com/17993169/151808953-f3966da5-c390-48f8-a86c-04d2b98add9a.MOV

Reply:
The flow for this is as follows:

1. User loads TL.
2. Apache (or whatever) asks for the auth cookie
3. If the user doesn't have it, the user is told to authenticate
4. (For OpenID/SAML) The user is redirected to the provider
5. The provider authenticates the user
6. The provider redirects back to the application to an endpoint that Apache consumes but not TL
7. Apache parses the data given to it
8. Apache gives the user the cookie if all the conditions are met (user has right role, etc)
9. User loads original page, is authenticated
10. Apache looks at user cookie and passes user into to TL via header

Reply:
By mod_auth_openidc (by far the most popular OpenID implementation for Apache), the redirect URL must be under the protected URL.